{{cookiecutter.repo_name}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,37 @@
+image: alpine
```

Comment:
So I think this would be some sort of compromise between your project's specific needs and what'd be general purpose.

Is there any reason you don't use an ubuntu image? Since that's what travis uses it would be a natural thing to use for users already used to travis.

Reply:
Another thing about the base image, in a more serious setup you'd use a custom image to avoid the overhead of installing tox and pip every time. Not sure what's the best way to tackle this. Maybe there's already a "fixed up and maintained" ubuntu image that could be used.

Reply:
I know literally nothing about alpine, except that https://gitlab.com/pages/sphinx used it as the base image for whatever reason.

You make a good point about efficiency...now that I look a little more, https://devopstutodoc.readthedocs.io/en/latest/hebergeurs/gitlab/gitlab.html uses an image that at least already has Python installed.

Reply:
> Another thing about the base image, in a more serious setup you'd use a custom image to avoid the overhead of installing tox and pip every time. Not sure what's the best way to tackle this. Maybe there's already a "fixed up and maintained" ubuntu image that could be used.

I have not been able to find any such thing, so ultimately I gave up and just made my own on DockerHub. It's very, very simple. https://hub.docker.com/r/dahanna/python.3.7-git-tox-alpine/dockerfile

I don't know why such a thing didn't already exist, it seems simple and obvious, but there it is.